llpc/lower/llpcSpirvLowerRayTracing.cpp

```diff
@@ -145,8 +151,14 @@ template <> void SpirvLowerRayTracing::createRayTracingFunc<OpTraceRayKHR>(Funct
 
     // Copy payload variable to the global payload variable
     auto payloadArg = func->getArg(TraceRayParam::Payload);
-    const DataLayout &dataLayout = m_module->getDataLayout();
-    unsigned payloadArgSize = alignTo(dataLayout.getTypeAllocSize(payloadArg->getType()->getPointerElementType()), 4);
+    auto payloadTypeArg = func->arg_end() - 1;
+    unsigned payloadArgSize = alignTo(m_module->getDataLayout().getTypeAllocSize(payloadTypeArg->getType()), 4);
+    // TODO: Remove this when LLPC will switch fully to opaque pointers.
+    if (!payloadArg->getType()->isOpaquePointerTy()) {
+      assert(payloadArgSize == (alignTo(m_module->getDataLayout().getTypeAllocSize(
```

Comment:
You could just write `assert(payloadArg->getType->isOpaquePointerTy() || payloadArgSize == ...);`

Reply:
Nice, thanks !